src/services/verifier.rs

```diff
@@ -455,10 +455,15 @@ fn verify_revealed_attribute_value(
     proof: &Presentation,
     attr_info: &RevealedAttributeInfo,
 ) -> Result<()> {
-    let reveal_attr_encoded = attr_info.encoded.to_string();
-    let reveal_attr_encoded = Regex::new("^0*")
-        .unwrap()
-        .replace_all(&reveal_attr_encoded, "");
+    let leading_zero_regex = Regex::new("^0[0-9]+").unwrap();
+    let encoded_info = &attr_info.encoded;
+
+    let reveal_attr_encoded = if leading_zero_regex.captures(encoded_info).is_some() {
+        encoded_info.trim_start_matches('0')
```

Comment:
I could not use the captured group sadly as it captures the entire number, seemes like an insignificant performance decrease to me.

Reply:
You could create a capture group for this I think?

Not sure if it's better for performance, or if it's better at all. But this would work I think:

```
^(0+)[1-9]([0-9]+)?$
```

Reply:
Actually the regex probably should use $ to avoid stripping leading zeros on "01.0" for instance

Reply:
Moved from the regex to `String::parse<i32>` as it changes `-0100` to `-100`.